### ISSUE TITLE
Ship stuck with impossible destination after hyperspace link disappears

### ISSUE THREAD

**Issue by jpretori** [endless-sky/endless-sky#4362]
**Describe the bug**
When Syndicate space gets severed from the Republic, one of my disabled ships was left stuck in Ankaa, with Diphda set as its destination. 

**To Reproduce**
1. Leave disabled ship in Ankaa, before Syndicate space is severed from the Republic.
2. Make sure the disabled ship has `"destination system" Diphda` in the save file
3. Do the missions to get the jump link severance to happen (disabled ship still stuck in Ankaa)
4. Go rescue the disabled ship
5. Observe that it is stuck in an infinite loop of landing on and then launching from the planet

**Expected behavior**
Remove impossible destinations from player ships when they become impossible.
OR, equally effective
Remove impossible destination from ship when it tries to navigate there

**System (please complete the following information):**
 - OS: Ubuntu 18 LTS
 - Game Source: Ubuntu package repositories
 - Version: "0.9.8"

**Comment by tehhowch** [comment]
Please reproduce with the latest official release, 0.9.9

**Comment by ChamEV** [comment]
This save was made in 0.98, and I just tested it in 0.99 and it exhibits this behavior.  Unknown whether this state is possible to reach now in 0.99.

[Broken Escorts~2 escorts stuck.txt](https://github.com/endless-sky/endless-sky/files/3289046/Broken.Escorts.2.escorts.stuck.txt)

I fixed it for my pilot at the time by removing the destination from each of the escorts.

system Deneb
"destination system" "Terra Incognita"

This case may be slightly different from ops however it exhibits the same behavior.
